### ISSUE TITLE
Create a new DS called `Fluid dictionary`

### ISSUE THREAD

**Issue by TonyTroeff** [TryAtSoftware/Extensions#45]
This should be the polymorphic brother of the standard dictionary.

This is the interface we should implement:
```C#
public interface IFluidDictionary<TKey>
{
    bool Set<T>(TKey key, T value);
    bool TryGetValue<T>(TKey key, out T value);
    T GetValueOrDefault<T>(TKey key);
    T GetRequiredValue<T>(TKey key);
    IEnumerable<TKey> Keys { get; }
}
```

**Comment by postmeback** [comment]
Just adding this interface will address this issue?

**Comment by TonyTroeff** [comment]
It should be implemented as well.

**Comment by postmeback** [comment]
I have cloned the repo. 

Can you please suggest to me the path where should I add the interface?

**Comment by TonyTroeff** [comment]
The interface should exist within the `TryAtSoftware.Extensions.Collection/Interfaces` folder.

**Comment by TonyTroeff** [comment]
You can refer to the [`Contributing`](https://github.com/TryAtSoftware/Extensions/blob/main/CONTRIBUTING.md#submitting-pull-requests) guide for concrete instructions

**Comment by postmeback** [comment]
1. Fork the repository to your own GitHub account - **Done**
2. Create a new branch: git checkout -b your-branch-name - **feat/45-Add-Fluid-Dictionary**
3. Make your desired changes, adhering to the [guidelines and best practices](https://github.com/TryAtSoftware/Extensions/blob/main/CONTRIBUTING.md#guidelines-and-best-practices); - **Done**
4. Commit your changes with clear and descriptive messages; - **Done**
5. Push your branch to your forked repository: git push origin your-branch-name - **Here I am getting stuck.**


Can you check from your side, if is there any permission-related issue?

**Comment by TonyTroeff** [comment]
You have not **forked** the repository - you have most likely **cloned** it.

**Comment by postmeback** [comment]
Yeah, sorry my bad. Anyways, can guide me towards the next steps?

**Comment by TonyTroeff** [comment]
At the root of the project will be OK, I suppose.

**Comment by postmeback** [comment]
You mean `src/FluidDictionary.cs` ?

**Comment by postmeback** [comment]
I have added a class with the required documentation. Please have a look and review. I will lupdate it accordingly
